dpe/src/x509.rs

```diff
@@ -1028,6 +1060,11 @@ mod tests {
         pub(crate) digest: &'a [u8],
     }
 
+    #[derive(asn1::Asn1Read)]
+    pub struct FwidList<'a> {
```

Comment:
I'm not sure this is right. I think it introduces an extra sequence around the fwidlist (because the asn1 crate treats structs as sequences)

```
DiceTcbInfo ::== SEQUENCE {
  ...
  fwids [6] IMPLICIT FWIDLIST OPTIONAL,
  ...
}


FWIDLIST ::== SEQUENCE SIZE (1..MAX) OF FWID
FWID ::== SEQUENCE {
  hashAlg OBJECT IDENTIFIER,
  digest OCTET STRING
}
```

I think this should create the hierarchy:

```
SEQUENCE // DiceTcbInfo
  SEQEUNCE OF // fwids field
    SEQUENCE // fwids[0]
    SEQUENCE // fwids[1]
    ...etc...
```

Reply:
Is it possible that asn1 crate doesn't treat `asn1::SequenceOf<'a, Fwid<'a>>` as an extra sequence if the contents is a struct (non primitive)?

When printout the raw asn1 from this test it seems correct. And asn1.parse_single is only able to parse the raw asn1 bytes with this struct definition.

Here's the ASN1 structure output from encode_tcb_info
```
SEQUENCE (3 elem)
  [6] (1 elem)
    SEQUENCE (2 elem)
      SEQUENCE (2 elem)
        OBJECT IDENTIFIER 2.16.840.1.101.3.4.2.1 sha-256 (NIST Algorithm)
        OCTET STRING (32 byte) BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
      SEQUENCE (2 elem)
        OBJECT IDENTIFIER 2.16.840.1.101.3.4.2.1 sha-256 (NIST Algorithm)
        OCTET STRING (32 byte) AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
  [8] (4 byte) FFFFFFFF
  [9] (4 byte) 11223344
```

Reply:
Reverted the TCBINFO changes